src/libraries/Common/tests/System/Net/Security/Kerberos/KerberosExecutor.cs

```diff
@@ -0,0 +1,143 @@
+
+    public const string DefaultAdminPassword = "PLACEHOLDERadmin.";
+
+    public const string DefaultUserPassword = "PLACEHOLDERcorrect20";
+
+    public KerberosExecutor(ITestOutputHelper testOutputHelper, string realm)
+    {
+        var krb5Config = Krb5Config.Default();
+        krb5Config.KdcDefaults.RegisterDefaultPkInitPreAuthHandler = false;
+
+        var logger = new KerberosDelegateLogger(
+            (level, categoryName, eventId, scopeState, logState, exception, log) =>
+                testOutputHelper.WriteLine($"[{level}] [{categoryName}] {log}")
+        );
+
+        _principalService = new FakePrincipalService(realm);
+
+        byte[] krbtgtPassword = new byte[16];
+        //RandomNumberGenerator.Fill(krbtgtPassword);
+
+        var krbtgt = new FakeKerberosPrincipal(PrincipalType.Service, "krbtgt", realm, krbtgtPassword);
+        _principalService.Add("krbtgt", krbtgt);
+        _principalService.Add($"krbtgt/{realm}", krbtgt);
+
+        _options = new ListenerOptions
+        {
+            Configuration = krb5Config,
+            DefaultRealm = realm,
+            RealmLocator = realm => new FakeRealmService(realm, krb5Config, _principalService),
+            Log = logger,
+            IsDebug = true,
+        };
+
+        _kdcListener = new FakeKdcServer(_options);
+        _realm = realm;
+        _servicePrincipals = new List<FakeKerberosPrincipal>();
+    }
+
+    public void Dispose()
+    {
+        _invokeHandle?.Dispose();
+        _kdcListener.Stop();
+        File.Delete(_krb5Path);
+        File.Delete(_keytabPath);
+    }
+
+    public void AddService(string name, string password = DefaultAdminPassword)
+    {
+        var principal = new FakeKerberosPrincipal(PrincipalType.Service, name, _realm, Encoding.Unicode.GetBytes(password));
+        _principalService.Add(name, principal);
+        _servicePrincipals.Add(principal);
+    }
+ 
+    public void AddUser(string name, string password = DefaultUserPassword)
+    {
+        var principal = new FakeKerberosPrincipal(PrincipalType.User, name, _realm, Encoding.Unicode.GetBytes(password));
+        _principalService.Add(name, principal);
+        _principalService.Add($"{name}@{_realm}", principal);
+    }
+
+    public async Task Invoke(Action method)
+    {
+        await PrepareInvoke();
+        _invokeHandle = RemoteExecutor.Invoke(method);
+    }
+
+    public async Task Invoke(Func<Task> method)
+    {
+        await PrepareInvoke();
+        _invokeHandle = RemoteExecutor.Invoke(method);
+    }
+
+    private async Task PrepareInvoke()
+    {
+        // Start the KDC server
+        var endpoint = await _kdcListener.Start();
+
+        // Generate krb5.conf
+        _krb5Path = Path.GetTempFileName();
+        File.WriteAllText(_krb5Path,
+            OperatingSystem.IsLinux() ?
+            $"[realms]\n{_options.DefaultRealm} = {{\n  master_kdc = {endpoint}\n  kdc = {endpoint}\n}}\n" :
```

Comment:
Why this depends on OS?

Reply:
Different notation to force TCP over UDP and to specify the KDC as master.